src/main/src/main/java/org/geoserver/config/GeoServerLoader.java

```diff
@@ -206,6 +219,8 @@ public void reload() throws Exception {
         
         loadCatalog( catalog, xp );
         loadGeoServer( geoserver, xp);
+        
+        reloadInitializers(geoserver);
```

Comment:
Order is correct, same as is done when initially loading the GeoServer configurion. I origionally tried with reloadInitializers first, but it turns out loadCatalog is responsible for creating a new ResourcePool() object.